superset-frontend/src/hooks/apiResources/tables.ts

```diff
@@ -117,6 +117,13 @@ const tableApi = api.injectEndpoints({
       }),
     }),
     tableMetadata: builder.query<TableMetaData, FetchTableMetadataQueryParams>({
+      providesTags: result =>
+        result
+          ? [
+              { type: 'TableMetadatas', id: result.name },
```

Comment:
Having a guaranteed unique id would be better. From what I can tell we would need to return it from the API though, so I just went with the name for now.

Reply:
what is `result.name` in this context? Is it a fully-qualified table name? Regardless I think at this time changing anything in the table selector flushes the metadata section, so even just a table name should be safe for now. If we were to support tables from multiple schemas to remain accessible in this metadata section, this could in theory create some collision.

Reply:
It's just the table name, no extra qualifier like schema. And I see it the same way regarding collisions, so I'm good leaving it as is it.